[b-10] add edit movie

Jokeroff opened the pull request.

Jokeroff committed:
[global fix] : aligning with database new column names

Jokeroff committed:
post construct

Jokeroff committed:
[s-1]: security token service
- SecurityService

Jokeroff committed:
[s-1]: security token service

Jokeroff committed:
[s-1] security token service : fixes
- AuthService reworked
- tests for Service and Controller
- ITests renamed
- redundant constructor removed from Movie

Jokeroff committed:
[lg-2] : add user and requestId logging
- draft of Interceptor added

Jokeroff committed:
[s-1]: security token service/GenreService.java
- Roles added
- some methods renamed
- dummy currencies added
- GlobalExceptionHandler improvement

Jokeroff committed:
[lg-2]: add user and requestId logging

Jokeroff committed:
Merge branch 's-1_security_token_service' into lg-2_add_user_and_requestId_logging

Jokeroff committed:
[lg-2] : renaming fix

Jokeroff committed:
[b-8]: add movie review

Jokeroff committed:
[b-9] : get all countries

Jokeroff committed:
[b-9]: unused code removed

Jokeroff committed:
[b-10]: add, edit movie

juliamanunia commented on src/main/java/com/lebediev/movieland/entity/Movie.java:
```diff
@@ -3,36 +3,38 @@
+    private int id;
+    private String nameRussian;
+    private String nameNative;
+    private int yearOfRelease;
```
why not LocalDate?

juliamanunia commented on src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java:
```diff
@@ -30,9 +30,17 @@
     }
 
+    @Override
+    public void add(Movie movie) {
```
how genres and countries can be added?

Jokeroff replied:
added in last commit

juliamanunia commented on src/main/java/com/lebediev/movieland/service/MovieServiceImpl.java:
```diff
@@ -30,9 +30,17 @@
 
+    @Override
+    public void update(Movie movie) {
+        movieDao.update(movie);
```
genres and countries

juliamanunia commented on src/main/java/com/lebediev/movieland/web/controller/ReviewController.java:
```diff
@@ -0,0 +1,45 @@
+    public void add(@RequestHeader String uuid, @RequestBody String json){
+        LOG.info("Start saving review for /review for uuid: {}", uuid);
+        UserToken userToken = authService.authorize(UUID.fromString(uuid));
+        if( userToken.isUser()){
```
admin also can add review

Jokeroff replied:
not requested

juliamanunia commented on src/main/java/com/lebediev/movieland/web/controller/interceptor/Interceptor.java:
```diff
@@ -0,0 +1,49 @@
+        if (uuid != null) {
+            currentUser = authorize(uuid);
+        }
+        MDC.put("requestId", UUID.randomUUID().toString());
```
why random? User's uuid should be there

Jokeroff replied:
this is request id, not user's

juliamanunia commented on src/main/java/com/lebediev/movieland/web/controller/utils/JsonConverter.java:
```diff
@@ -2,57 +2,102 @@
         }
     }
+
+    public static String toJson(TokenDto tokenDto) {
```
to many static overloaded methods. Maybe divide a job will be better?

Jokeroff replied:
made some simplification

juliamanunia commented on src/test/java/com/lebediev/movieland/dao/jdbc/integrationtests/JdbcMovieDaoITest.java:
```diff
@@ -80,10 +80,47 @@ public void testGetMoviesByGenreIdOrdered() {
+        actual.setYearOfRelease(2000);
+        jdbcMovieDao.add(actual);
+        int id = jdbcTemplate.queryForObject("SELECT id FROM movie WHERE nameNative = 'before update' AND yearOfRelease = 2000", Integer.class);
+        assertNotEquals(id, null);
```
when id will be null?

Jokeroff replied:
good point - if no rows selected then exception will be thrown, fixed, thanks )

Jokeroff committed:
[s-1, b-8] fixes:
- return Review
- Auth Service sanity changes

Jokeroff committed:
[b-10]: add edit movie
- added adding/update countries and genres
- MovieDtoForUpdate

Jokeroff committed:
[b-10] add edit movie
- adding of countries ang genres
- MovieDtoForUpdate
- JsonConverter simplification
